functions/Set-DbaCompression.ps1

```diff
@@ -0,0 +1,508 @@
+                                     + @CompressionTypeRecommendation + ',SORT_IN_TEMPDB=ON)';
+
+                              IF DATEDIFF(mi, @StartTime, CURRENT_TIMESTAMP) < @MaxRunTimeInMinutes
+                                 BEGIN
+									PRINT 'Compressing table/index: '
+                                    + @Schema + '.' + @TableName;
+                                    EXEC sp_executesql
+                                    @SQL;
+
+                                    Update ##setdbacompression
+                                    SET     AlreadyProcessed = 1
+                                    WHERE   PK = @PK;
+
+                                    SET @CompressedCount = @CompressedCount
+                                    + 1;
+                                 END
+                              ELSE
+                                 BEGIN
+                                       PRINT 'Max runtime reached. Some compression performed. Exiting...';
+                                       BREAK
+                                 END
+
+                              FETCH cCompress INTO @Schema, @TableName, @Partition, @IndexName,
+                                    @IndexType, @CompressionTypeRecommendation, @PK;
+                        END
+
+                  CLOSE cCompress;
+                  DEALLOCATE cCompress;
+
+                  SELECT 
+                   PK				
+                  ,DBName = DB_Name()
+				  ,[Schema] 
+				  ,[TableName] 
+				  ,[IndexName] 
+				  ,[Partition] 
+				  ,[IndexID] 
+				  ,[IndexType] 
+				  ,[PercentScan] 
+				  ,[PercentUpdate] 
+				  ,[RowEstimatePercentOriginal] 
+				  ,[PageEstimatePercentOriginal]
+				  ,[CompressionTypeRecommendation] 
+				  ,SizeCurrentKB = [SizeCurrent]
+				  ,SizeRequestedKB = [SizeRequested]
+                  ,PercentCompression
+				  ,AlreadyProcessed
+				  FROM ##setdbacompression
+                  WHERE AlreadyProcessed=1;
+
+                  IF OBJECT_ID('tempdb..##setdbacompression' , 'U') IS NOT NULL
+                  DROP TABLE ##setdbacompression
+				
+                  IF OBJECT_ID('tempdb..##tmpEstimateRow' , 'U') IS NOT NULL
+                  DROP TABLE ##tmpEstimateRow
+
+                  IF OBJECT_ID('tempdb..##tmpEstimatePage' , 'U') IS NOT NULL
+                  DROP TABLE ##tmpEstimatePage;"
+	}
+	
+	process {
+		
+		foreach ($instance in $SqlInstance) {
+			try {
+				Write-Message -Level VeryVerbose -Message "Connecting to $instance" -Target $instance
+				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SourceSqlCredential -MinimumVersion 10
+			}
+			catch {
+				Stop-Function -Message "Failed to process Instance $Instance" -ErrorRecord $_ -Target $instance -Continue
+			}
+			
+            $Server.ConnectionContext.StatementTimeout = 0
+                
+                if ($Server.EngineEdition -eq 'Standard' -and $Server.VersionMajor -lt '13')
```

Comment:
Would you mind adding a comment to just explain above this line why you validate in the connection for version 10 but then check for less than 13 here? Since this is due to MS changes in Standard Edition I'm assuming, worth just noting that to avoid confusion and make sure it is not removed.